tensorflow_privacy/privacy/membership_inference_attack/seq2seq_mia.py

```diff
@@ -0,0 +1,360 @@
+  batch_correct_preds = 0.0
+  batch_total_preds = 0.0
+  for sequence_logits, sequence_labels in zip(batch_logits, batch_labels):
+    preds = sparse_categorical_accuracy(K.constant(sequence_labels),
+                                        K.constant(sequence_logits))
+    batch_correct_preds += preds.numpy().sum()
+    batch_total_preds += len(sequence_labels)
+
+  return batch_correct_preds, batch_total_preds
+
+
+def create_seq2seq_attacker_data(attack_input_data: Seq2SeqAttackInputData,
+                                 privacy_report_metadata: PrivacyReportMetadata,
+                                 test_fraction: float = 0.25,
+                                 balance: bool = True) -> AttackerData:
+  """Prepares Seq2SeqAttackInputData to train ML attackers.
+
+  Uses logits and losses to generate ranks and performs a random train-test
+  split.
+
+  Also computes metadata (loss, accuracy) for the model under attack
+  and populates respective fields of PrivacyReportMetadata.
+
+  Args:
+    attack_input_data: Original Seq2SeqAttackInputData
+    privacy_report_metadata: the metadata of the model under attack.
+    test_fraction: Fraction of the dataset to include in the test split.
+    balance: Whether the training and test sets for the membership inference
+      attacker should have a balanced (roughly equal) number of samples from the
+      training and test sets used to develop the model under attack.
+
+  Returns:
+    AttackerData.
+  """
+  attack_input_train, loss_train, accuracy_train = _get_attack_features_and_metadata(attack_input_data.logits_train,
+                                                                                     attack_input_data.labels_train)
+  attack_input_test, loss_test, accuracy_test = _get_attack_features_and_metadata(attack_input_data.logits_test,
+                                                                                  attack_input_data.labels_test)
+
+  if balance:
+    min_size = min(len(attack_input_train), len(attack_input_test))
+    attack_input_train = _sample_multidimensional_array(attack_input_train,
+                                                        min_size)
+    attack_input_test = _sample_multidimensional_array(attack_input_test,
+                                                       min_size)
+
+  features_all = np.concatenate((attack_input_train, attack_input_test))
+
+  # Reshape for classifying one-dimensional features
+  features_all = features_all.reshape(-1, 1)
+
+  labels_all = np.concatenate(
+      ((np.zeros(len(attack_input_train))), (np.ones(len(attack_input_test)))))
+
+  # Perform a train-test split
+  features_train, features_test, \
+  is_training_labels_train, is_training_labels_test = \
+    model_selection.train_test_split(
+        features_all, labels_all, test_size=test_fraction, stratify=labels_all)
+
+  # Populate accuracy, loss fields in privacy report metadata
+  privacy_report_metadata = privacy_report_metadata or PrivacyReportMetadata()
```

Comment:
Can we move this PrivacyReportMetadata() to the argument line?

So 

privacy_report_metadata: PrivacyReportMetadata = PrivacyReportMetadata()

instead of this line. Reassigning input arguments can lead to unintended consequences, plus it's difficult to do refactoring if we create default parameters in the function.